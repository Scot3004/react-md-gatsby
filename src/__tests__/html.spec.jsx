import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Html, { getInlinedStyles } from '../html';

it('renders correctly', () => {
  const tree = shallow(<Html />);
  expect(toJson(tree)).toMatchSnapshot();
});

describe('Mocking ENV Production', () => {
  let defaultEnv;

  beforeAll(() => {
    defaultEnv = global.process.env.NODE_ENV;
    global.process.env.NODE_ENV = "production"
  });

  afterAll(() => {
    global.process.env.NODE_ENV = defaultEnv;
  });

  it('renders correctly Production', () => {
    const tree = shallow(<Html />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});

describe('Mocking Console logging', () => {
  let defaultConsole;

  beforeAll(() => {
    defaultConsole = global.console.log;
    global.console.log = jest.fn();
  });

  afterAll(() => {
    global.console.log = defaultConsole;
  });

  it('error on file', () => {
    jest.mock('../public/styles.css', () => {
      throw new Error('Loading error');
    });
    getInlinedStyles()
    expect(global.console.log).toBeCalled();
  });
});
