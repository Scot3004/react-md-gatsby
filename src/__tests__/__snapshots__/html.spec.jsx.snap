// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mocking ENV Production renders correctly Production 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charSet="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport"
    />
    <link
      href={Object {}}
      rel="shortcut icon"
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": Object {},
        }
      }
      id="gatsby-inlined-css"
    />
  </head>
  <body>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      id="___gatsby"
    />
  </body>
</html>
`;

exports[`renders correctly 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charSet="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport"
    />
    <link
      href={Object {}}
      rel="shortcut icon"
    />
  </head>
  <body>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      id="___gatsby"
    />
  </body>
</html>
`;
